TLSharp.Core/Network/TcpTransport.cs

```diff
@@ -32,27 +33,29 @@ public TcpTransport(string address, int port, TcpClientConnectionHandler handler
             }
         }
 
-        public async Task Send(byte[] packet)
+        public async Task Send(byte[] packet, CancellationToken token)
         {
             if (!_tcpClient.Connected)
                 throw new InvalidOperationException("Client not connected to server.");
 
-            var tcpMessage = new TcpMessage(sendCounter, packet);
+            var tcpMessage = new TcpMessage(_sendCounter, packet);
 
-            await _stream.WriteAsync(tcpMessage.Encode(), 0, tcpMessage.Encode().Length);
-            sendCounter++;
+            await _stream.WriteAsync(tcpMessage.Encode(), 0, tcpMessage.Encode().Length, token).ConfigureAwait(false);
+            _sendCounter++;
         }
 
-        public async Task<TcpMessage> Receive()
+        public async Task<TcpMessage> Receive(CancellationToken token)
```

Comment:
this two methods above are public and still don't have default token

Reply:
@IaRuslan I've just realized that the same happens in Authenticator.cs, MtProtoPlainSender.cs, and MtProtoSender.cs

Reply:
ping